pkg/metrics/metrics.go

```diff
@@ -17,12 +17,62 @@ limitations under the License.
 package metrics
 
 import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
 	"github.com/prometheus/client_golang/prometheus"
 	dto "github.com/prometheus/client_model/go"
 
 	"k8s.io/kube-state-metrics/pkg/options"
 )
 
+// Metric represents a single line entry in the /metrics export format
+type Metric string
+
+// NewMetric returns a new Metric
+func NewMetric(name string, labelKeys []string, labelValues []string, value float64) (*Metric, error) {
+	if len(labelKeys) != len(labelValues) {
+		return nil, errors.New("expected labelKeys to be of same length as labelValues")
+	}
+
+	m := ""
+
+	m = m + name
+
+	m = m + labelsToString(labelKeys, labelValues)
+
+	m = m + fmt.Sprintf(" %v", value)
+
+	m = m + "\n"
```

Comment:
nit: just create one `m := name + labelsToString(...) + fmt.Sprintf(" %g", value)` line.
note the `%g` suggestion which is the default for `%v` in conjunction with floats and seems to be more explicit.